### PR TITLE
PWGHF: Add McCollisionId to the event table of treeCreatorLcToPKPi

### PR DESCRIPTION
**Pull request by DelloStritto**
Adding McCollisionId to the event table to match it with the MC particle table.
@qgp @vkucera

### CODE REVIEW

File: PWGHF/TableProducer/treeCreatorLcToPKPi.cxx [hunk 1]
```diff
@@ -80,6 +80,7 @@ DECLARE_SOA_COLUMN(OriginMcGen, originMcGen, int8_t);
 DECLARE_SOA_COLUMN(IsCandidateSwapped, isCandidateSwapped, int8_t);
 DECLARE_SOA_INDEX_COLUMN_FULL(Candidate, candidate, int, HfCand3Prong, "_0");
 // Events
+DECLARE_SOA_INDEX_COLUMN(McCollision, mccollision);
```
Comment (vkucera) [inline PWGHF/TableProducer/treeCreatorLcToPKPi.cxx:83]:
```suggestion
DECLARE_SOA_INDEX_COLUMN(McCollision, mcCollision);
```

File: PWGHF/TableProducer/treeCreatorLcToPKPi.cxx [hunk 2]
```diff
@@ -203,8 +205,13 @@ struct HfTreeCreatorLcToPKPi {
     // Filling event properties
     rowCandidateFullEvents.reserve(collisions.size());
     for (const auto& collision : collisions) {
+      auto mcCollisionId = 0;
+      if (collision.has_mcCollision()) {
+        mcCollisionId = collision.mcCollisionId();
+      }
```
Comment (vkucera) [inline PWGHF/TableProducer/treeCreatorLcToPKPi.cxx:211]: Do we need to save the collisions that do not have a matching MC collision?

**Review by vkucera — CHANGES_REQUESTED** [review]

**Review by vkucera — DISMISSED** [review]
Thanks @DelloStritto 
@qgp What do you think?

File: PWGHF/TableProducer/treeCreatorLcToPKPi.cxx [hunk 3]
```diff
@@ -203,8 +205,12 @@ struct HfTreeCreatorLcToPKPi {
     // Filling event properties
     rowCandidateFullEvents.reserve(collisions.size());
     for (const auto& collision : collisions) {
+      if (!collision.has_mcCollision()) {
+        continue;
```
Comment (qgp) [inline PWGHF/TableProducer/treeCreatorLcToPKPi.cxx:209]: @DelloStritto Do we really want to silently drop these? Wouldn't it make sense to keep them and store -1 for the MC collision ID?

**Comment by qgp** [comment]
Looks good to me. Only question about silently dropping collisions without an MC collision (see inline comment.

File: PWGHF/TableProducer/treeCreatorLcToPKPi.cxx (hunk 3 above)
Comment (vkucera) [reply PWGHF/TableProducer/treeCreatorLcToPKPi.cxx:209]: @qgp Where would we use these collisions without a matching reconstruction level?

**Review by vkucera — APPROVED** [review]

**Pull request merged**